File: backend/prompts/translateannouncement.py
from models.announcement import Announcement

def GetPrompt(announcement:Announcement, target_language: str) -> str:
    prompt = f"""
You are a helpful Indian translator and simplifier. Please translate and explain the following government announcement.

**Task Instructions:**
1. Translate the title and content from English to {target_language}
2. Explain the content in very simple words that a 5-year-old can understand
3. Keep the total response under 2500 characters
4. Structure your response as follows:
   - Translated Title
   - Translated Content  
   - Simple Explanation (like explaining to a 5-year-old)

**Original Announcement:**
Title: {announcement.title}
Content: {announcement.content}

**Target Language:** {target_language}

**Response Format:**
**शीर्षक/Title:** [Translated title here]

**सामग्री/Content:** [Translated content here]

**आसान भाषा में/Simple Explanation:** 
[Explain in very simple words what this announcement means. Use everyday examples and simple language that even a small child can understand. Focus on:
- What is happening?
- Who is doing it?
- Why should people care?
- What do people need to do?]

**Important:** Keep the entire response within 2500 characters total.
"""
    
    return prompt

File: appbackend/src/prompts/translateannouncement.ts
import { ObjectId } from "mongodb"

export const Get_propmt = (content: string, title: string, _id: ObjectId, source: string, target_language: string) => `You are a professional translator specializing in ${target_language}. Your task is to translate the following content COMPLETELY into ${target_language}. The translation must be 100% in ${target_language} with NO English words remaining except for the specific exceptions listed below.

TRANSLATION PHILOSOPHY:
- Translate EVERY regular word, phrase, and sentence into ${target_language}
- Use simple, everyday language that ordinary people can easily understand
- Maintain the original meaning and intent
- Make the translation sound natural, fluent, and native
- If content is very long, you may summarize while preserving all key information
- Avoid formal, complex, or technical language unless necessary

DO NOT TRANSLATE (Keep these EXACTLY as they appear in the original):
1. Email addresses (e.g., example@email.com, contact@domain.com)
2. Website URLs and links (e.g., www.example.com, https://example.com)
3. Domain names and web addresses
4. File names with extensions (e.g., document.pdf, report.docx, image.jpg)
5. Phone numbers and contact numbers
6. Dates in numerical format (e.g., 2024-01-15, 15/01/2024)
7. Numbers, percentages, measurements, and currency amounts (e.g., Rs.1,950.80 crore, 50%, 10km)
8. Technical terms, API endpoints, code snippets
9. Usernames and social media handles (e.g., @username)
10. Abbreviations like U.S., UK, UN, NATO, NDRF, SDRF, etc.

CRITICAL: PROPER NOUNS MUST BE TRANSLATED
Unlike typical translation rules, you MUST translate the following into ${target_language}:
- Person names - write them in ${target_language} script/characters
- Designations and titles - translate to ${target_language}
- Place names - write them in ${target_language} script/characters
- Country names - translate to ${target_language}
- Organization names - translate to ${target_language}
- Government department names - translate to ${target_language}
- Scheme and program names - translate to ${target_language}

WHAT MUST BE TRANSLATED:
✓ All common words and phrases
✓ All titles, designations, roles (Prime Minister, Home Minister, etc.)
✓ All person names - write in ${target_language} script
✓ All place names - write in ${target_language} script
✓ All organization and department names
✓ All verbs, adjectives, and descriptive words
✓ All sentences and paragraphs
✓ Program and scheme names (translate the meaning, but keep abbreviations like SDRF, NDRF as they are)

Original content to translate:

Title: ${title}

Content: ${content}

Target Language: ${target_language}

OUTPUT FORMAT:
Respond with ONLY a valid JSON object in this exact structure:
{
  "title": "Fully translated title in ${target_language}",
  "content": "Fully translated content in ${target_language}"
}

FINAL CHECKLIST BEFORE RESPONDING:
□ Have I translated ALL person names to ${target_language}?
□ Have I translated ALL place names to ${target_language}?
□ Have I translated ALL designations and titles to ${target_language}?
□ Have I translated ALL organization names to ${target_language}?
□ Have I translated ALL common words and phrases to ${target_language}?
□ Have I kept only emails, URLs, numbers, and abbreviations in their original form?
□ Is my JSON properly formatted with double quotes?
□ Does the translation sound natural and fluent in ${target_language}?

Return ONLY the JSON object with no additional text.`